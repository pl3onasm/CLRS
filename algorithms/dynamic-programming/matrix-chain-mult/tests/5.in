10 9 6 6 5 2 2 2 1 8 7 7 15 6 2 7 8 8 8