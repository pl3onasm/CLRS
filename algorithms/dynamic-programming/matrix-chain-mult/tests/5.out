Min cost: 818

An optimal parenthesization:
((A1 x (A2 x (A3 x (A4 x (A5 x (A6 x (A7 x A8))))))) x (((((((((A9 x A10) x A11) x A12) x A13) x A14) x A15) x A16) x A17) x A18))
